domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt

```diff
@@ -154,6 +181,12 @@ class ExplorationProgressController @Inject constructor(
             )
           }
         } finally {
+          // If the answer was submitted on behalf of the Continue interaction, don't save
+          // checkpoint because it will be saved when the learner moves to the next state.
+          if (answerOutcome.state.interaction.id != "Continue") {
```

Comment:
Why don't we need to save it for continue state? I'm a bit confused here.

Reply:
We are not saving for a checkpoint for continue interaction after the answer is submitted on behalf of continue interaction and the exploration goes into viewing state, the stateFragment moves to the next state without any user interaction. So saving here as we do in any other interaction will cause us to save one unnecessary checkpoint.